### DISCUSSION TITLE
SIWE as an alternative to SMTP (email) signUp

### DISCUSSION THREAD

**Discussion by adapole** [mastodon/mastodon#33045] [General]
I came across an article by Justin Garrison, [Mastodon is Doomed](https://justingarrison.com/blog/2023-04-24-mastodon-is-doomed/)

> It’s only a matter of time until there will be a CVE found in the official Mastodon software which will leave a vast majority of instances vulnerable. 
> If the instance you’re on is compromised you will lose your identity because private keys are stored as part of the instance. No private keys, no identity, no portability.

I'm proposing an alternative to the current "_username, password_" login method, where there won't be a need for a password when using [Sign-In With Ethereum](https://docs.login.xyz/). TLDR: `SIWE` is a **new** form of `authentication` that enables users to control their digital identity.

SIWE, are more secure than passwords because the server/instance only stores the `PublicKey` and the `Privatekey` remains with the user in their wallet, therefore they are resistant to phishing, always strong, and have no shared secrets.

What are your thoughts on adding SIWE feature to mastodon?

**Comment by ghost** [comment]
Love it as long as we have a secure secret generation function

**Comment by MzTea1** [comment]
Glad you love it. And. I don’t know what is it.

**Reply by ghost** [reply]
The secret generation function is what makes the keys. If it makes keys that are predictable/crackable or too small to be safer than it puts account in danger of being taken, but this is safer than storing private keys on sever. This goes into key generation in terms of encryption  https://en.wikipedia.org/wiki/Key_generation

**Comment by MzTea1** [comment]
Lol interesting emails again
